splink/internals/linker_components/visualisations.py

```diff
@@ -75,9 +75,12 @@ class LinkerVisualisations:
     def __init__(self, linker: Linker):
         self._linker = linker
 
-    def match_weights_chart(self) -> ChartReturnType:
+    def match_weights_chart(self, as_dict: bool = True) -> ChartReturnType:
```

Comment:
`as_dict` should be `False` by default.

i.e. all of the functions in `linker.visualiastions` should return an Altair chart by default, but if the user passes `as_dict=True` they can get back the dict representation without going through altair 

I think this is also the reasons for the test failures

Reply:
That is what I intended but must have made a typo, it is updated now when you want to look again